v2/internal/frontend/desktop/windows/go-webview2/webviewloader/env_create.go

```diff
@@ -0,0 +1,159 @@
+func CreateCoreWebView2EnvironmentWithOptions(environmentCompletedHandler ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler, opts ...option) error {
+	var params environmentOptions
+	for _, opt := range opts {
+		opt(&params)
+	}
+
+	var err error
+	var dllPath string
+	var runtimeType webView2RunTimeType
+	if browserExecutableFolder := params.browserExecutableFolder; browserExecutableFolder != "" {
+		runtimeType = webView2RunTimeTypeRedistributable
+		dllPath, err = findEmbeddedClientDll(browserExecutableFolder)
+	} else {
+		runtimeType = webView2RunTimeTypeInstalled
+		dllPath, _, err = findInstalledClientDll(params.preferCanary)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return createWebViewEnvironmentWithClientDll(dllPath, runtimeType, params.userDataFolder,
+		&params, environmentCompletedHandler)
+}
+
+func createWebViewEnvironmentWithClientDll(lpLibFileName string, runtimeType webView2RunTimeType, userDataFolder string,
+	envOptions *environmentOptions, envCompletedHandler ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler) error {
+
+	if !filepath.IsAbs(lpLibFileName) {
+		return fmt.Errorf("lpLibFileName must be absolute")
+	}
+
+	dll, err := windows.LoadDLL(lpLibFileName)
+	if err != nil {
+		return fmt.Errorf("Loading DLL failed: %w", err)
+	}
+
+	defer func() {
+		canUnloadProc, err := dll.FindProc("DllCanUnloadNow")
+		if err != nil {
+			return
+		}
+
+		if r1, _, _ := canUnloadProc.Call(); r1 != windows.NO_ERROR {
+			return
+		}
+
+		dll.Release()
+	}()
+
+	createProc, err := dll.FindProc("CreateWebViewEnvironmentWithOptionsInternal")
+	if err != nil {
+		return fmt.Errorf("Unable to find CreateWebViewEnvironmentWithOptionsInternal entrypoint: %w", err)
+	}
+
+	userDataPtr, err := windows.UTF16PtrFromString(userDataFolder)
+	if err != nil {
+		return err
+	}
+
+	envOptionsCom := combridge.New2[iCoreWebView2EnvironmentOptions, iCoreWebView2EnvironmentOptions2](
+		envOptions, envOptions)
+
+	defer envOptionsCom.Close()
+
+	envCompletedHandler = &environmentCreatedHandler{envCompletedHandler}
+	envCompletedCom := combridge.New[iCoreWebView2CreateCoreWebView2EnvironmentCompletedHandler](envCompletedHandler)
+	defer envCompletedCom.Close()
+
+	const unknown = 1
+	hr, _, err := createProc.Call(
+		uintptr(unknown),
+		uintptr(runtimeType),
+		uintptr(unsafe.Pointer(userDataPtr)),
+		uintptr(envOptionsCom.Ref()),
+		uintptr(envCompletedCom.Ref()))
+
+	if hr != 0 {
+		if err == nil || err == windows.ERROR_SUCCESS {
+			err = syscall.Errno(hr)
+		}
+		return err
+	}
+
+	return nil
+}
+
+type environmentCreatedHandler struct {
+	originalHandler ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler
+}
+
+func (r *environmentCreatedHandler) EnvironmentCompleted(errorCode HRESULT, createdEnvironment *ICoreWebView2Environment) HRESULT {
+	// The OpenWebview2Loader has some retry logic and retries once, didn't encounter any case when this would have been
```

Comment:
Perhaps @jchv has more insight

Reply:
Unfortunately, I do not. I also never noticed a case where it is needed. If there *is* a good reason for it, one would probably have to peel back more layers to find it.

Reply:
Thanks for your reply. So I think we could leave this as it is for now and see what the feedback will be from the users.

Reply:
Thanks for taking the time to leave a comment @jchv - appreciate it 🙏